torch/_dynamo/eval_frame.py

```diff
@@ -627,26 +627,40 @@ def guard_export_print(guards):
 
 
 @dataclasses.dataclass
-class Constraint:
+class ConstraintTarget:
```

Comment:
The new name looks a bit confusing. What does it mean?

Reply:
It's the target of a constraint. The target is a (tensor, dimension) pair, basically. With this change I factored out the fields of `Constraint` into the target part and the actual constraint part (range), which also allows me to share the constraint part with another target.